internal/servicelog/ringbuffer.go

```diff
@@ -57,6 +62,10 @@ func NewRingBuffer(size int) *RingBuffer {
 	return &rb
 }
 
+// DoAtomicDiscard causes the buffer to track boundaries between Write calls ensuring that when
+// full, bytes are discarded
+func (rb *RingBuffer) DoAtomicDiscard() { rb.atomicDiscard = true }
```

Comment:
I'd probably call this `EnableAtomicDiscard`, as "do" implies it's going to do an atomic discard right now.

That said, I'm wondering if we can find a better name than "atomic discard", and/or explain it more clearly in the doc comment here. I *think* I understand what it does, but it's a bit vague here.